libsolidity/formal/Predicate.cpp

```diff
@@ -336,7 +347,9 @@ optional<string> Predicate::expressionToString(smtutil::Expression const& _expr,
 	if (smt::isArray(*_type))
 	{
 		auto const& arrayType = dynamic_cast<ArrayType const&>(*_type);
-		solAssert(_expr.name == "tuple_constructor", "");
+		if (_expr.name != "tuple_constructor")
+			return {};
```

Comment:
Why was this change made?

Reply:
Because this is the branch that builds the value for an array. Solidity arrays are represented as a SMT tuple (elements, length), whose value is represented by a `tuple_constructor` operation. However, in the case of the tx data, the `blockhash` member is represented directly by an SMT array which won't have a `tuple_constructor`. Since we don't know the length, we can't generically use `fillArray` the same way that is there right now. I think in a following PR we could assume length 256 (because of https://docs.soliditylang.org/en/v0.8.0/units-and-global-variables.html?highlight=blockhash#block-and-transaction-properties) there and then call `fillArray`, if that `if` is true and the expression is `blockhash`.